sources/core/Xenko.Core.Mathematics/Int2.cs

```diff
@@ -195,12 +195,11 @@ public int LengthSquared()
         /// </summary>
         public void Normalize()
         {
-            int length = Length();
-            if (length > MathUtil.ZeroTolerance)
+            if (X!=0 && Y!=0)
```

Comment:
That's incorrect. A vector with one component set to zero still has a normalized version. In that case it is simply 1 in the other component.

Reply:
yes, just replace '&&' with "||"